examples/tv-casting-app/android/App/app/src/main/jni/com/chip/casting/TvCastingApp.java

```diff
@@ -136,30 +146,119 @@ public void discoverVideoPlayerCommissioners(
       multicastLock.acquire();
 
       nsdManager = (NsdManager) applicationContext.getSystemService(Context.NSD_SERVICE);
+      List<DiscoveredNodeData> discoveredPlayers = new ArrayList<>();
       nsdDiscoveryListener =
           new NsdDiscoveryListener(
               nsdManager,
               DISCOVERY_TARGET_SERVICE_TYPE,
               DISCOVERY_TARGET_DEVICE_TYPE_FILTER,
               preCommissionedVideoPlayers,
-              discoverySuccessCallback,
+              new SuccessCallback<DiscoveredNodeData>() {
+                @Override
+                public void handle(DiscoveredNodeData commissioner) {
+                  Log.d(TAG, "Discovered commissioner added " + commissioner);
+                  discoveredPlayers.add(commissioner);
+                  discoverySuccessCallback.handle(commissioner);
+                }
+              },
               discoveryFailureCallback,
               nsdManagerResolverAvailState);
 
       nsdManager.discoverServices(
           DISCOVERY_TARGET_SERVICE_TYPE, NsdManager.PROTOCOL_DNS_SD, nsdDiscoveryListener);
       Log.d(TAG, "TvCastingApp.discoverVideoPlayerCommissioners started");
+
+      this.reportSleepingVideoPlayerCommissionersFuture =
+          Executors.newScheduledThreadPool(1)
+              .schedule(
+                  () -> {
+                    Log.d(
+                        TAG,
+                        "Scheduling reportSleepingCommissioners with commissioner count "
+                            + (preCommissionedVideoPlayers != null
+                                ? preCommissionedVideoPlayers.size()
+                                : 0));
+                    reportSleepingVideoPlayerCommissioners(
+                        preCommissionedVideoPlayers, discoveredPlayers, discoverySuccessCallback);
+                  },
+                  CHIP_DEVICE_CONFIG_STR_DISCOVERY_DELAY_SEC * 1000,
+                  TimeUnit.MILLISECONDS);
+
       this.discoveryStarted = true;
     }
   }
 
+  private void reportSleepingVideoPlayerCommissioners(
```

Comment:
it will be worthwhile adding documentation here on what you're doing, the logic isn't really obvious on why we're doing this

Reply:
Done, Added docs above the ReportSleepingCommissioners() function in both Java and C++.